packages/edit-site/src/components/page-patterns/search-items.js

```diff
@@ -20,11 +20,14 @@ import {
 	PATTERN_DEFAULT_CATEGORY,
 	PATTERN_USER_CATEGORY,
 	PATTERN_TYPES,
+	TEMPLATE_PART_POST_TYPE,
 } from '../../utils/constants';
 
 // Default search helpers.
-const defaultGetName = ( item ) => item.name || '';
-const defaultGetTitle = ( item ) => item.title;
+const defaultGetName = ( item ) =>
+	item.type !== TEMPLATE_PART_POST_TYPE ? item.name || '' : '';
```

Comment:
[Reference about names](https://github.com/WordPress/gutenberg/pull/62927/files#r1658553606).

It seems the `name` is used to search items by rank here. This change avoids targeting the `name` of a template part while still using it for searching patterns.

Given the name is the slug, it sounds like it shouldn't be used to search items. Though, I reckon I'm not familiar with how we do this operation in other parts of the app and can be convinced one way or the other. What I find odd is the inconsistency of using it for searching patterns but not parts.

Reply:
The name doesn't exist for template parts, in trunk it uses the id which doesn't make sense. For patterns names are more meaningful in a sense.

Reply:
Yeah, I saw the other thread, the data I extracted was based on what this PR actually removes (`templateToPatterns`), so this is good 👍